src/python/gudhi/tensorflow/perslay.py

```diff
@@ -0,0 +1,284 @@
+    """
+    def __init__(self, samples, theta, **kwargs):
+        """
+        Constructor for the FlatPerslayPhi class.
+  
+        Parameters:
+            samples (float numpy array): grid elements on which to evaluate the constant functions, of the form [x_1, ..., x_n].
+            theta (float): sigmoid parameter used to approximate the constant function with a differentiable sigmoid function. The bigger the theta, the closer to a constant function the output will be. 
+        """
+        super().__init__(dynamic=True, **kwargs)
+        self.samples = tf.Variable(initial_value=samples, trainable=True)
+        self.theta   = tf.Variable(initial_value=theta,   trainable=True)
+        
+    def build(self, input_shape):
+        return self
+        
+    def call(self, diagrams):
+        """
+        Apply FlatPerslayPhi on a ragged tensor containing a list of persistence diagrams.
+
+        Parameters:
+            diagrams (n x None x 2): ragged tensor containing n persistence diagrams. The second dimension is ragged since persistence diagrams can have different numbers of points.
+
+        Returns:
+            output (n x None x num_samples): ragged tensor containing the evaluations on the 1D grid of the 1D constant functions corresponding to the persistence diagram points. The second dimension is ragged since persistence diagrams can have different numbers of points.
+            output_shape (int numpy array): shape of the output tensor.
+        """
+        samples_d = tf.expand_dims(tf.expand_dims(self.samples,0),0)
+        xs, ys = diagrams[:,:,0:1], diagrams[:,:,1:2]
+        output = 1./(1.+tf.math.exp(-self.theta*(.5*(ys-xs)-tf.math.abs(samples_d-.5*(ys+xs)))))
+        output_shape = self.samples.shape
+        return output, output_shape
+
+class Perslay(tf.keras.layers.Layer):
+    """
+    This is a TensorFlow layer for vectorizing persistence diagrams in a differentiable way within a neural network. This function implements the PersLay equation, see `the corresponding article <http://proceedings.mlr.press/v108/carriere20a.html>`_.
+    """
+    def __init__(self, weight, phi, perm_op, rho, **kwargs):
+        """
+        Constructor for the Perslay class.
+  
+        Parameters:
+            weight (function): weight function for the persistence diagram points. Can be either :class:`~gudhi.tensorflow.GridPerslayWeight`, :class:`~gudhi.tensorflow.GaussianMixturePerslayWeight`, :class:`~gudhi.tensorflow.PowerPerslayWeight`, or a custom function.
+            phi (function): transformation function for the persistence diagram points. Can be either :class:`~gudhi.tensorflow.GaussianPerslayPhi`, :class:`~gudhi.tensorflow.TentPerslayPhi`, :class:`~gudhi.tensorflow.FlatPerslayPhi`, or a custom function.
```

Comment:
I guess I should look at the documentation of one of the existing Phi classes to find out what is expected of a custom function?

Reply:
Yes, I added a comment about this in the doc.